test/test-clusters/src/main/java/org/elasticsearch/test/cluster/local/DefaultSettingsProvider.java

```diff
@@ -70,17 +76,20 @@ public Map<String, String> get(LocalNodeSpec nodeSpec) {
             .stream()
             .filter(LocalNodeSpec::isMasterEligible)
             .map(LocalNodeSpec::getName)
+            .filter(Objects::nonNull)
             .collect(Collectors.joining(","));
 
         if (masterEligibleNodes.isEmpty()) {
-            throw new IllegalStateException(
-                "Cannot start cluster '" + nodeSpec.getCluster().getName() + "' as it configured with no master-eligible nodes."
+            LOGGER.warn(
+                "Cluster configured with no master-eligible nodes. It may cause split-brain if more than one node is in the cluster."
             );
         }
 
-        settings.put("cluster.initial_master_nodes", "[" + masterEligibleNodes + "]");
-        settings.put("discovery.seed_providers", "file");
-        settings.put("discovery.seed_hosts", "[]");
+        if (masterEligibleNodes.isEmpty() == false) {
+            settings.put("cluster.initial_master_nodes", "[" + masterEligibleNodes + "]");
+            settings.put("discovery.seed_providers", "file");
+            settings.put("discovery.seed_hosts", "[]");
+        }
```

Comment:
@mark-vieira @breskeby 
This is something I would definitely like your opinion on. I am now allowing creating cluster without setting the `cluster.initial_master_nodes` where it was explicitly disallowed previously. I believe this is OK when there is a single node, but it possible can create some problems in more complex tests. 
I need that to be able to start the node without the explicit name in the `JsonLogsFormatAndParseIT` test.

Reply:
Well, in the case where the node doesn't have a name, `masterEligibleNodes` here would be an empty string, no? Would setting `initial_master_nodes` to `[]` be equivalent to not setting it at all or is the behavior different?

Reply:
Alternatively, could we make that above error condition more specific. That is, only throw an exception when there are no master eligible nodes _and_ the cluster contains multiple nodes?